tensorflow_lite_support/cc/task/vision/image_transformer.cc

```diff
@@ -0,0 +1,195 @@
+#include "tensorflow_lite_support/metadata/metadata_schema_generated.h"
+
+namespace tflite {
+namespace task {
+namespace vision {
+
+namespace {
+
+using ::absl::StatusCode;
+using ::tflite::metadata::ModelMetadataExtractor;
+using ::tflite::support::CreateStatusWithPayload;
+using ::tflite::support::StatusOr;
+using ::tflite::support::TfLiteSupportStatus;
+using ::tflite::task::core::AssertAndReturnTypedTensor;
+using ::tflite::task::core::TaskAPIFactory;
+using ::tflite::task::core::TfLiteEngine;
+
+}  // namespace
+
+/* static */
+StatusOr<std::unique_ptr<ImageTransformer>> ImageTransformer::CreateFromOptions(
+    const ImageTransformerOptions& options,
+    std::unique_ptr<tflite::OpResolver> resolver) {
+  RETURN_IF_ERROR(SanityCheckOptions(options));
+
+  // Copy options to ensure the ExternalFile outlives the constructed object.
+  auto options_copy = absl::make_unique<ImageTransformerOptions>(options);
+
+  std::unique_ptr<ImageTransformer> image_transformer;
+  //TODO: Should be model_file_with_metadata?
+  if (options_copy->model_file_with_metadata()) {
+    ASSIGN_OR_RETURN(
+        image_classifier,
+        TaskAPIFactory::CreateFromExternalFileProto<ImageTransformer>(
+            &options_copy->model_file_with_metadata(), std::move(resolver),
+            options_copy->num_threads(), options_copy->compute_settings()));
+  } else if (options_copy->base_options().has_model_file()) {
+    ASSIGN_OR_RETURN(image_classifier,
+                     TaskAPIFactory::CreateFromBaseOptions<ImageTransformer>(
+                         &options_copy->base_options(), std::move(resolver)));
+  } else {
+    // Should never happen because of SanityCheckOptions.
+    return CreateStatusWithPayload(
+        StatusCode::kInvalidArgument,
+        absl::StrFormat("Expected exactly one of `base_options.model_file` or "
+                        "`model_file_with_metadata` to be provided, found 0."),
+        TfLiteSupportStatus::kInvalidArgumentError);
+  }
+
+  RETURN_IF_ERROR(image_transformer->Init(std::move(options_copy)));
+
+  return image_transformer;
+}
+
+/* static */
+absl::Status ImageTransformer::SanityCheckOptions(
+    const ImageTransformerOptions& options) {
+  int num_input_models = (options.base_options().has_model_file() ? 1 : 0) +
+                         (options.has_model_file_with_metadata() ? 1 : 0);
+
+  if (num_input_models != 1) {
+    return CreateStatusWithPayload(
+        StatusCode::kInvalidArgument,
+        absl::StrFormat("Expected exactly one of `base_options.model_file` or "
+                        "`model_file_with_metadata` to be provided, found %d.",
+                        num_input_models),
+        TfLiteSupportStatus::kInvalidArgumentError);
+  }
+  if (options.base_options().compute_settings().tflite_settings().cpu_settings().num_threads() == 0 ||
+      options.base_options().compute_settings().tflite_settings().cpu_settings().num_threads() < -1) {
```

Comment:
Is this the correct way to get `num_threads`?

Reply:
Yes, that's correct. But you don't need to check num_threads specifically. It has been verified [here](https://github.com/tensorflow/tflite-support/blob/765e82607c6a85ac9cdbda12467ff1f56826914f/tensorflow_lite_support/cc/task/core/task_api_factory.h#L133-L141) when the base task is created.